pkg/service/aws/cost.go

```diff
@@ -145,6 +145,132 @@ func (svc AWSController) GetWorkspacesCost(ctx context.Context, req *proto.GetWo
 	return costResponse, nil
 }
 
+func (svc AWSController) GetApplicationsCost(ctx context.Context, req *proto.GetApplicationsCostRequest) (*proto.GetApplicationsCostResponse, error) {
+
+	session, err := NewSession(ctx, "", req.GetAccountName())
+
+	if err != nil {
+		svc.logger.Errorw("can't start AWS session", "error", err)
+		return nil, err
+	}
+
+	stsClient := session.getSTSClient()
+
+	callerIdentity, err := stsClient.GetCallerIdentity(nil)
+
+	if err != nil {
+		svc.logger.Errorw("failed to get identity", "error", err)
+		return nil, err
+	}
+
+	accound_id := callerIdentity.Account
+
+	svc.logger.Debugw("fetched accountId", "id", accound_id)
+
+	filter := &costexplorer.Expression{
+		And: []*costexplorer.Expression{
+
+			{
+				Dimensions: &costexplorer.DimensionValues{
+					Key:    aws.String("LINKED_ACCOUNT"),
+					Values: aws.StringSlice([]string{*accound_id}),
+				},
+			},
+			{
+				Tags: &costexplorer.TagValues{
+					Key:    aws.String(constants.WorkspaceId),
+					Values: aws.StringSlice(req.GetApplicationIds()),
+				},
+			},
+
+			{
+
+				Not: &costexplorer.Expression{
+					Dimensions: &costexplorer.DimensionValues{
+						Key: aws.String("RECORD_TYPE"),
+						Values: aws.StringSlice([]string{
+							"Credit",
+						}),
+					},
+				},
+			},
+		},
+	}
+
+	input := costexplorer.GetCostAndUsageInput{
```

Comment:
this can be function too, passing only arg which it needs. simplifies read a lot.